### ISSUE TITLE
Allow splitting of result zip files into smaller segments to allow for network upload restrictions

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#850]
**Reported by simalam on 15 Nov 39291161 00:53 UTC**
Due to the increasing size of some of the project files, especially the climateprediction.net Project, users behind proxy servers are unable to finalise the upload of completed projects as zip file is too large. This is seen as "HTTP errors" in the messages log as per below:
  16/04/2009 11:34:21 AM    climateprediction.net   Started upload of hadam3p_mabq_1971_2_006045576_1_1.zip[ 16/04/2009 11:34:24 AM climateprediction.net   Temporarily failed upload of hadam3p_mabq_1971_2_006045576_1_1.zip: HTTP error[[BR]([BR]])]
  16/04/2009 11:34:24 AM    climateprediction.net   Backing off 2 hr 22 min 36 sec on upload of hadam3p_mabq_1971_2_006045576_1_1.zip[is happening is that this zip file is 25.21MB in size and the HTTP upload request (not sure if it is a HTTP PUT or POST) is blocked by a proxy server as it is larger than the proxy server max HTTP upload size. See my post in this [http://climateapps2.oucs.ox.ac.uk/cpdnboinc/forum_thread.php?id=6520&nowrap=true#36498 thread at climateprediction.net]([BR]]

What) for more info.

Could a network preference be created that splits the upload zip files into definable segments, e.g. per MB? Maybe similar to the trickle upload process?

This would allow users who are restricted in their internet usage, especially in a work or academic environment to continue to participate in large projects such as the climateprediction.net.

Cheers, [[BR]]
Sim Alam

Migrated-From: http://boinc.berkeley.edu/trac/ticket/873

**Comment by romw** [comment]
**Commented by simalam on 3 Dec 39291194 19:06 UTC**
If this could be applied to existing/pending transfers that would be even better.

**Comment by romw** [comment]
**Commented by Ageless on 12 Sep 39291749 02:40 UTC**
Adjusted title.

**Comment by romw** [comment]
**Commented by simalam on 8 Jun 40720273 03:06 UTC**
Any updates on this? It has been nearly 18 months since I logged it.

Cheers,
Sim

**Comment by romw** [comment]
**Commented by davea on 5 Feb 40720364 13:20 UTC**
Please set <http_debug>,
so I can see the HTTP error return.
Maybe there's some way to detect that the request was too large

**Comment by romw** [comment]
**Commented by simalam on 17 Jun 40720654 19:06 UTC**
Ok, from the below log file the HTTP error return appears to be:
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: HTTP/1.0 413 Request Entity Too Large```


Extract of stdoutdae.txt with server names etc changed:
  ```20-Sep-2010 14:11:03 [---](ID#29]) [HTTP_OP::libcurl_exec(): ca-bundle 'D:\Program Files\BOINC\ca-bundle.crt'```
  ```20-Sep-2010 14:11:03 [---](http]) [HTTP_OP::libcurl_exec(): ca-bundle set```
  ```20-Sep-2010 14:11:03 [---](http]) [HTTP_OP::no_proxy_for_url(): http://uploader.oerc.ox.ac.uk/cpdn_cgi/file_upload_handler```
  ```20-Sep-2010 14:11:03 [---](proxy]) [returning false```
  ```20-Sep-2010 14:11:03 [---](proxy]) [setting up proxy my.proxy.com:8080```
  ```20-Sep-2010 14:11:03 [climateprediction.net](proxy]:) Started upload of hadam3p_ndh0_1968_2_006165950_2_1.zip```
  ```20-Sep-2010 14:11:04 [[http](---]) [Info:  timeout on name lookup is not supported```
  ```20-Sep-2010 14:11:04 [---](ID#29]) [[ID#29](http]) Info:  About to connect() to proxy my.proxy.com port 8080 (#0)```
  ```20-Sep-2010 14:11:04 [[http](---]) [Info:    Trying 172.26.254.49... ```
  ```20-Sep-2010 14:11:04 [---](ID#29]) [[ID#29](http]) Info:  Connected to my.proxy.com (172.26.254.49) port 8080 (#0)```
  ```20-Sep-2010 14:11:04 [[http](---]) [Sent header to server: POST http://uploader.oerc.ox.ac.uk/cpdn_cgi/file_upload_handler HTTP/1.1```
  ```20-Sep-2010 14:11:04 [---](ID#29]) [[ID#29](http]) Sent header to server: User-Agent: BOINC client (windows_x86_64 6.11.7)```
  ```20-Sep-2010 14:11:04 [[http](---]) [Sent header to server: Host: uploader.oerc.ox.ac.uk```
  ```20-Sep-2010 14:11:04 [---](ID#29]) [[ID#29](http]) Sent header to server: Accept: */*```
  ```20-Sep-2010 14:11:04 [[http](---]) [Sent header to server: Accept-Encoding: deflate, gzip```
  ```20-Sep-2010 14:11:04 [---](ID#29]) [[ID#29](http]) Sent header to server: Proxy-Connection: Keep-Alive```
  ```20-Sep-2010 14:11:04 [[http](---]) [Sent header to server: Content-Type: application/x-www-form-urlencoded```
  ```20-Sep-2010 14:11:04 [---](ID#29]) [[ID#29](http]) Sent header to server: Content-Length: 290```
  ```20-Sep-2010 14:11:04 [[http](---]) [Sent header to server: ```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Info:  HTTP 1.0, assume close after body```
  ```20-Sep-2010 14:11:05 [[http](---]) [Received header from server: HTTP/1.0 200 OK```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Received header from server: Date: Mon, 20 Sep 2010 05:11:05 GMT```
  ```20-Sep-2010 14:11:05 [[http](---]) [Received header from server: Server: Apache/2.2.13 (Linux/SUSE)```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Received header from server: Content-Type: text/plain```
  ```20-Sep-2010 14:11:05 [[http](---]) [Received header from server: X-Cache: MISS from l2-13.proxy.com```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Received header from server: X-Cache-Lookup: MISS from l2-13.proxy.com:8079```
  ```20-Sep-2010 14:11:05 [[http](---]) [Received header from server: X-Cache: MISS from proxy02.my.proxy.com```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Received header from server: X-Cache-Lookup: MISS from proxy02.my.proxy.com:8080```
  ```20-Sep-2010 14:11:05 [[http](---]) [Received header from server: Via: 1.0 l2-13.proxy.com:8079 (squid/2.6.STABLE20), 1.0 proxy02.my.proxy.com:8080 (squid/2.6.STABLE20)```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Received header from server: Proxy-Connection: close```
  ```20-Sep-2010 14:11:05 [[http](---]) [Received header from server: ```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http_xfer]) HTTP: wrote 93 bytes```
  ```20-Sep-2010 14:11:05 [[http](---]) [Info:  Expire cleared```
  ```20-Sep-2010 14:11:05 [---](ID#29]) [[ID#29](http]) Info:  Closing connection #0```
  ```20-Sep-2010 14:11:06 [[http](---]) HTTP_OP::libcurl_exec(): ca-bundle set```
  ```20-Sep-2010 14:11:06 [[proxy](---]) HTTP_OP::no_proxy_for_url(): http://uploader.oerc.ox.ac.uk/cpdn_cgi/file_upload_handler```
  ```20-Sep-2010 14:11:06 [[proxy](---]) returning false```
  ```20-Sep-2010 14:11:06 [[proxy](---]): setting up proxy my.proxy.com:8080```
  ```20-Sep-2010 14:11:06 [[http](---]) [Info:  timeout on name lookup is not supported```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Info:  About to connect() to proxy my.proxy.com port 8080 (#0)```
  ```20-Sep-2010 14:11:06 [[http](---]) [Info:    Trying 172.26.254.49... ```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Info:  Connected to my.proxy.com (172.26.254.49) port 8080 (#0)```
  ```20-Sep-2010 14:11:06 [[http](---]) [Sent header to server: POST http://uploader.oerc.ox.ac.uk/cpdn_cgi/file_upload_handler HTTP/1.1```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Sent header to server: User-Agent: BOINC client (windows_x86_64 6.11.7)```
  ```20-Sep-2010 14:11:06 [[http](---]) [Sent header to server: Host: uploader.oerc.ox.ac.uk```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Sent header to server: Accept: */*```
  ```20-Sep-2010 14:11:06 [[http](---]) [Sent header to server: Accept-Encoding: deflate, gzip```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Sent header to server: Proxy-Connection: Keep-Alive```
  ```20-Sep-2010 14:11:06 [[http](---]) [Sent header to server: Content-Type: application/x-www-form-urlencoded```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Sent header to server: Content-Length: 26416862```
  ```20-Sep-2010 14:11:06 [[http](---]) [Sent header to server: Expect: 100-continue```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Sent header to server:   ```20-Sep-2010 14:11:06 [[http](---]) [Info:  Expire cleared```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Info:  HTTP 1.0, assume close after body```
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: HTTP/1.0 413 Request Entity Too Large```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Received header from server: Server: squid/2.6.STABLE20```
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: Date: Mon, 20 Sep 2010 04:11:06 GMT```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Received header from server: Content-Type: text/html```
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: Content-Length: 1379```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Received header from server: Expires: Mon, 20 Sep 2010 04:11:06 GMT```
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: X-Squid-Error: ERR_TOO_BIG 0```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Received header from server: X-Cache: MISS from proxy02.my.proxy.com```
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: X-Cache-Lookup: NONE from proxy02.my.proxy.com:8080```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Received header from server: Via: 1.0 proxy02.my.proxy.com:8080 (squid/2.6.STABLE20)```
  ```20-Sep-2010 14:11:06 [[http](---]) [Received header from server: Proxy-Connection: close```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Received header from server: ```
  ```20-Sep-2010 14:11:06 [[http_xfer](---]) [HTTP: wrote 1379 bytes```
  ```20-Sep-2010 14:11:06 [---](ID#29]) [[ID#29](http]) Info:  Expire cleared```
  ```20-Sep-2010 14:11:06 [[http](---]) [Info:  Closing connection #0```
  ```20-Sep-2010 14:11:07 [climateprediction.net](ID#29]) Temporarily failed upload of hadam3p_ndh0_1968_2_006165950_2_1.zip: HTTP error```
  ```20-Sep-2010 14:11:07 [climateprediction.net] Backing off 9 hr 18 min 38 sec on upload of hadam3p_ndh0_1968_2_006165950_2_1.zip```

**Comment by romw** [comment]
**Commented by davea on 24 Dec 40722535 10:40 UTC**
Apache has configuration parameters for max request size:

LimitXMLRequestBody 134217728
LimitRequestBody 134217728 

These apparently default to 1MB;
the above lines set them to 128MB

**Comment by romw** [comment]
**Commented by simalam on 21 Nov 40722822 02:40 UTC**
Replying to [davea](comment:7):
> Apache has configuration parameters for max request size:
> 
> LimitXMLRequestBody 134217728
> LimitRequestBody 134217728 
> 
> These apparently default to 1MB;
> the above lines set them to 128MB

I don't understand how that helps.

The problem I have is that the BOINC client has to connect to the internet via a proxy server. It is the proxy server that is generating the HTTP 413 errors and I can't get the upload size limit increased on it.

**Comment by romw** [comment]
**Commented by davea on 18 Mar 40722855 16:00 UTC**
Yes, it won't help that.
I'll keep this ticket open but won't have time to work on it for a while.

**Comment by ChristianBeer** [comment]
Splitting upload in multiple segments is not trivial. Todays proxy servers should support a higher upload limit. If this feature is still needed please open a new issue.
